packages/react-sdk/src/components/image.tsx

```diff
@@ -1,24 +1,187 @@
 import React, { forwardRef, type ElementRef, type ComponentProps } from "react";
+import { warnOnce } from "../utils/warn-once";
 
 const defaultTag = "img";
 
-type ImageProps = ComponentProps<typeof defaultTag>;
+type ImageProps = ComponentProps<typeof defaultTag> & { quality?: number };
+
+/**
+ * max(...imageSizes) must be less then min(...deviceSizes)
+ * idea is to support images with vw size less than 100vw
+ * https://github.com/vercel/next.js/blob/canary/packages/next/shared/lib/image-config.ts
+ **/
+const imageSizes = [16, 32, 48, 64, 96, 128, 256, 384];
+const deviceSizes = [640, 750, 828, 1080, 1200, 1920, 2048, 3840];
+
+const allSizes = [...imageSizes, ...deviceSizes];
+
+/**
+ * https://developers.cloudflare.com/images/image-resizing/url-format/
+ **/
+const cloudflareImageLoader = ({
+  width,
+
+  src,
+  quality,
+}: {
+  width: number;
+  src: string;
+  quality?: number;
+}) => {
+  let resizeOrigin = "";
+
+  if (process.env.NODE_ENV !== "production") {
+    if (!allSizes.includes(width)) {
+      warnOnce("Width must be only from allowed values");
+    }
+
+    // No image transformation in development
+    resizeOrigin = "https://webstudio.is";
+  }
+
+  const params = [
+    `width=${width}`,
+    quality != null && `quality=${quality}`,
+    // TODO: Must be checked how it works with caching (like avif image is cached and then read by the browser without avif support)
+    "format=auto",
+  ].filter(Boolean);
+
+  return `${resizeOrigin}/cdn-cgi/image/${params.join(",")}/${src}`;
```

Comment:
So in production `resizeOrigin` will be `""`?

Reply:
Yes, as for every host we have image resize enabled for now.

Reply:
This forces no dns lookup, + ability to push high-priority images using http2

Reply:
See https://blog.cloudflare.com/early-hints/